app/Controllers/AutoriController.php

```
@@ -136,17 +156,206 @@ public function update(Request $request, Response $response, mysqli $db, int $id
+                    $dest = $dir . '/' . $name;
+                    // Re-encode through GD to strip any embedded payload (mirrors cover handling).
+                    if ($this->reencodeAuthorPhoto($bytes, $detected, $dest)) {
+                        @chmod($dest, 0644);
+                        $stored = '/uploads/autori/' . $name;
+                        return [
+                            'foto' => $stored,
+                            'deleteOnSuccess' => ($existing !== '' && $existing !== $stored) ? $existing : null,
+                            'deleteOnFailure' => $stored, // remove this fresh upload if the DB write fails
+                        ];
+                    }
+                    \App\Support\SecureLogger::error('Author photo re-encode failed (unsupported or corrupt image)');
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\SecureLogger' in method 'resolveAuthorPhoto'. (undefined)

(StaticAccess)